ingestor/adx/tasks.go

```diff
@@ -540,26 +600,66 @@ func applySubstitutions(body, startTime, endTime string, clusterLabels map[strin
 }
 
 func (t *SummaryRuleTask) submitRule(ctx context.Context, rule v1.SummaryRule, startTime, endTime string) (string, error) {
+	// Track operation duration
+	start := time.Now()
+	var status string
+	defer func() {
+		duration := time.Since(start).Seconds()
+		metrics.IngestorKustoOperationDuration.WithLabelValues(
+			rule.Spec.Database, "submit_rule", status).Observe(duration)
+		metrics.IngestorSummaryRuleSubmissions.WithLabelValues(
+			rule.Spec.Database, rule.Namespace, rule.Name, status).Inc()
+	}()
+
 	// NOTE: We cannot do something like `let _startTime = datetime();` as dot-command do not permit
 	// preceding let-statements.
 	body := applySubstitutions(rule.Spec.Body, startTime, endTime, t.ClusterLabels)
 
+	// Add timeout to prevent indefinite blocking on Kusto operations
+	ctx, cancel := context.WithTimeout(ctx, 5*time.Minute)
+	defer cancel()
+
 	// Execute asynchronously
 	stmt := kql.New(".set-or-append async ").AddUnsafe(rule.Spec.Table).AddLiteral(" <| ").AddUnsafe(body)
 	res, err := t.kustoCli.Mgmt(ctx, stmt)
 	if err != nil {
+		status = "error"
 		return "", fmt.Errorf("failed to execute summary rule %s.%s: %w", rule.Spec.Database, rule.Name, err)
 	}
 
-	return operationIDFromResult(res)
+	operationId, err := operationIDFromResult(res)
+	if err != nil {
+		status = "error"
+		return "", err
+	}
+
+	status = "success"
+	logger.Infof("Successfully submitted summary rule %s/%s to database %s, operation ID: %s",
```

Comment:
Can remove these additional success logs but thought I'd get feedback from others first as they could be useful in the early stages while any bugs are ironed out.